libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -925,6 +1070,102 @@ impl NodeAPI for Greenlight {
         }
     }
 
+    async fn max_amount_to_send(
+        &self,
+        payee_node_id: Option<Vec<u8>>,
+        max_hops: u32,
+        last_hop_hints: Option<&RouteHintHop>,
+    ) -> NodeResult<(Vec<MaxChannelAmount>, PaymentPath)> {
+        let mut client = self.get_node_client().await?;
+
+        // Consider the hints as part of the route. If there is a routing hint we will
+        // attmept to calculate the path untill the last hop in the hint and then add
+        // the last hop to the path.
+        let (last_node, max_hops) = match last_hop_hints {
+            Some(hop) => (hex::decode(&hop.src_node_id)?, max_hops - 1),
+            None => match payee_node_id.clone() {
+                Some(node_id) => (node_id, max_hops),
+                None => {
+                    return Err(NodeError::RouteNotFound(anyhow!(
+                        "no payee node id or last hop hints provided, cannot calculate max amount"
+                    )));
+                }
+            },
+        };
+
+        // fetch a route from greenlight
+        let route_response = client
+            .get_route(GetrouteRequest {
+                id: last_node.clone(),
+                amount_msat: Some(pb::cln::Amount { msat: 1_000 }),
```

Comment:
I am not sure that it will work as the whole point of constructing a route manually is to bypass the node pathfinding that might return no route for amounts near local balance.
The assumption of this whole feature is that there exist one route with 3 hops that has enough liquidity to the destination so there shouldn't be a difference if we change the amount.
3 hops means: 
`gl -> LSP -> Routing Node -> Swapper`

Reply:
The base fee is a large factor in small payments, so it could play a role if there is an alternative route with lower base fees for example. I understand the pathfinding won't work too well when you get close to the balance. Maybe try with amount * 90% - 1000sat? Or the 1 sat if that ends up to be too low?

Reply:
As long as we get the route, Why should we prefer this route or another in the context of this function?
In the outer scope the way we use it we ensure we get the route we want because we are combining with the swapper routing hints.

Reply:
I see in the docs now that we can pass 0 so it ignores the amount and just lookup for a path. I think it makes more sense so will try that.

Reply:
I pushed using zero amount. This should also bypass the min_htlc limits.